cogs/annual_handover_and_reset.py

```diff
@@ -0,0 +1,198 @@
+                await handover_channel.set_permissions(
+                    committee_member,
+                    read_messages=True,
+                    send_messages=True,
+                )
+
+            logger.debug("Removing Committee role from user: %s", committee_member)
+            await committee_member.remove_roles(
+                committee_role,
+                reason=f"{ctx.user} used TeX-Bot slash-command: \"handover\"",
+            )
+
+        await initial_response.edit(
+            content=":hourglass: Giving committee-elect Committee role and "
+            "removing committee-elect... :hourglass:",
+        )
+
+        committee_elect_member: discord.Member
+        for committee_elect_member in committee_elect_role.members:
+            logger.debug("Giving user: %s, the committee role.", committee_elect_member)
+            await committee_elect_member.add_roles(
+                committee_role,
+                reason=f"{ctx.user} used TeX-Bot slash-command: \"handover\"",
+            )
+
+            logger.debug("Removing Committee-Elect role from user: %s", committee_elect_member)
+            await committee_elect_member.remove_roles(
+                committee_elect_role,
+                reason=f"{ctx.user} used TeX-Bot slash-command: \"handover\"",
+            )
+
+        await initial_response.edit(content=":white_check_mark: Handover procedure complete!")
+
+
+class AnnualResetRolesCommandCog(TeXBotBaseCog):
+    """Cog class that defines the reset_roles command."""
+
+    YEAR_ROLE_NAMES: Final[frozenset[str]] = frozenset(
+        {
+        "Foundation Year",
+        "First Year",
+        "Second Year",
+        "Final Year",
+        "Year In Industry",
+        "Year Abroad",
+        "PGT",
+        },
+    )
+
+    @discord.slash_command(  # type: ignore[no-untyped-call, misc]
+        name="annual_role_reset",
+        description="Removes member and year roles from all users",
+    )
+    @CommandChecks.check_interaction_user_has_committee_role
+    @CommandChecks.check_interaction_user_in_main_guild
+    async def annual_role_reset(self, ctx: TeXBotApplicationContext) -> None:
+        """
+        Definition & callback response of the "reset_roles" command.
+
+        The "reset_roles" command removes the "Memeber" and respective year roles from
+        any user that has them and subsequently resets the GroupMadeMember database model.
+        """
+        # NOTE: Shortcut accessors are placed at the top of the function, so that the exceptions they raise are displayed before any further errors may be sent
+        main_guild: discord.Guild = self.bot.main_guild
+        member_role: discord.Role = await self.bot.member_role
+
+        logger.debug("Reset roles command called.")
+        initial_response: discord.Interaction | discord.WebhookMessage = await ctx.respond(
+            ":hourglass: Resetting membership and year roles... :hourglass:",
```

Comment:
* Message content should be passed to `ctx.respond()` function as a named argument: `content="...",`
* Missing `ephemeral=True,` argument